src/Compilers/CSharp/Portable/Symbols/Synthesized/SynthesizedStaticConstructor.cs

```diff
@@ -371,5 +374,54 @@ internal override int CalculateLocalSyntaxOffset(int localPosition, SyntaxTree l
             var containingType = (SourceMemberContainerTypeSymbol)this.ContainingType;
             return containingType.CalculateSyntaxOffsetInSynthesizedConstructor(localPosition, localTree, isStatic: true);
         }
+
+        internal bool ShouldEmit(ImmutableArray<BoundInitializer> boundInitializersOpt = default)
+        {
+            if (_lazyShouldEmit.HasValue())
+            {
+                return _lazyShouldEmit.Value();
+            }
+
+            var shouldEmit = CalculateShouldEmit(boundInitializersOpt);
+            _lazyShouldEmit = shouldEmit.ToThreeState();
+            return shouldEmit;
+        }
+
+        private bool CalculateShouldEmit(ImmutableArray<BoundInitializer> boundInitializersOpt = default)
+        {
+            if (boundInitializersOpt.IsDefault)
+            {
+                if (!(ContainingType is SourceMemberContainerTypeSymbol sourceType))
+                {
+                    return true;
+                }
```

Comment:
Is this case for synthesized classes? Are we emitting a static constructor for those cases always?

Reply:
Based on the usages of SynthesizedStaticConstructor I think the only synthesized class to be concerned about is SynthesizedClosureEnvironment which has static fields for the display class and sometimes for the delegate.

Reply:
Consider asserting `ContainingType` is `SynthesizedClosureEnvironment` so we catch other cases as synthesized class types are added.

---
In reply to: [415115832](https://github.com/dotnet/roslyn/pull/43234#discussion_r415115832) [](ancestors = 415115832)